src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```
@@ -2798,6 +2904,37 @@ private void Remove(int startIndex, int count, out StringBuilder chunk, out int
             AssertInvariants();
         }
 
+        public Rune GetRuneAt(int index)
+        {
+            if (TryGetRuneAt(index, out Rune value))
+            {
+                return value;
+            }
+            throw new Exception($"Unable to get rune at {index}.");
+        }
```

We need to throw the correct exception, like `ArgumentException` and we need to store the exception message in the resources https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/Resources/Strings.resx.

---

or maybe `InvalidOperationException` make more sense here.

---

In `String.cs`, we have:
```cs
// Gets the character at a specified position.
//
[IndexerName("Chars")]
public char this[int index]
{
    [Intrinsic]
    get
    {
        if ((uint)index >= (uint)_stringLength)
            ThrowHelper.ThrowIndexOutOfRangeException();
        return Unsafe.Add(ref _firstChar, (nint)(uint)index /* force zero-extension */);
    }
}
```

So I replaced the highlighted code with:
```cs
ThrowHelper.ThrowIndexOutOfRangeException();
return default; // Necessary to prevent return value syntax error
```

The default IndexOutOfRange error message mentions an array, but this is the case for string as well so we should be fine:
```cs
Console.WriteLine("Hello World"[100]);
// Unhandled exception. System.IndexOutOfRangeException: Index was outside the bounds of the array.
```